### PR TITLE
DPE-1231 cluster set default setup

### PR DESCRIPTION
**Pull request by paulomach**
## Issue

Cluster set not being setup

## Solution

Setup cluster set as primary cluster on deployment

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -177,11 +178,13 @@ def _on_config_changed(self, _) -> None:
...
-            self.app_peer_data["cluster-name"] = (
-                self.config.get("cluster-name") or f"cluster_{generate_random_hash()}"
-            )
+        # Set the cluster and cluster-set name in the peer relation databag
+        # if it is not already set
+        for key in ("cluster-name", "cluster-set-name"):
+            if not self.app_peer_data.get(key):
+                self.app_peer_data[key] = (
+                    self.config.get(key) or f"cluster_{generate_random_hash()}"
+                )
```
[the remark this answers is not in the document]
Comment (paulomach) [reply src/charm.py:187]: yes. most commonly this should be manually set

File: src/charm.py [hunk 2]
```diff
@@ -177,11 +178,13 @@ def _on_config_changed(self, _) -> None:
...
             return
 
-        # Set the cluster name in the peer relation databag if it is not already set
-        if not self.app_peer_data.get("cluster-name"):
-            self.app_peer_data["cluster-name"] = (
-                self.config.get("cluster-name") or f"cluster_{generate_random_hash()}"
-            )
+        # Set the cluster and cluster-set name in the peer relation databag
+        # if it is not already set
+        for key in ("cluster-name", "cluster-set-name"):
```
[the remark this answers is not in the document]
Comment (paulomach) [reply src/charm.py]: 0184f86

File: config.yaml [hunk 3]
```diff
@@ -5,6 +5,9 @@ options:
   cluster-name:
     description: "Optional - Name of the MySQL InnoDB cluster"
     type: "string"
+  cluster-set-domain-name:
+    description: "Optional - Domain name for the MySQL InnoDB cluster set"
+    type: "string"
```
Comment (taurus-forever) [inline config.yaml:10]: Is the name approved by @delgod / Mohamed?
Should be ACKed quickly, but it is a procedure.

**Review by taurus-forever — APPROVED** [review]
LGTM, ping to @delgod to approve config option.

File: config.yaml (hunk 3 above)
Comment (paulomach) [reply config.yaml]: The direction is to remove the config altogether and use an auto-generated.

Comment (paulomach) [reply config.yaml]: b147c53

File: src/charm.py [hunk 4]
```diff
@@ -178,13 +178,12 @@ def _on_config_changed(self, _) -> None:
...
-            if not self.app_peer_data.get(key):
-                self.app_peer_data[key] = (
-                    self.config.get(key) or f"cluster_{generate_random_hash()}"
-                )
+        # Create and set cluster and cluster-set names in the peer relation databag
+        common_hash = generate_random_hash()
+        self.app_peer_data.setdefault(
+            "cluster-name", self.config.get("cluster-name", f"cluster-{common_hash}")
+        )
+        self.app_peer_data.setdefault("cluster-set-domain-name", f"cluster-set-{common_hash}")
```
[the remark this answers is not in the document]
Comment (paulomach) [reply src/charm.py:186]: Yes. They need not to be related and removing the config allow us to move forward for now.

**Pull request merged**